src/Aspire.Hosting.Azure.PostgreSQL/AzurePostgresExtensions.cs

```diff
@@ -17,6 +18,10 @@ namespace Aspire.Hosting;
 /// </summary>
 public static class AzurePostgresExtensions
 {
+    private const string ServerResourceVersion = "2023-03-01-preview";
```

Comment:
@davidfowl @mitchdenny @tg-msft - how should we think about resource versions? Should we always hard-code the version we expect, and then periodically update to new versions on some sort of cadence? Similar to our other dependency versions.

Reply:
At the moment I have everything default to "latest stable" in `Azure.Provisioning.*` (though I'm pulling the info from ARM so it's not perfect yet).  The long term plan is to have `Azure.Provisioning.*` automatically release whenever the management plane libraries are updated so the "latest stable" will change [seemingly at random.](https://www.nuget.org/packages?q=Azure.ResourceManager.)

As a general rule, moving to newer service versions is goodness for customers but there are occasionally breaking changes.  We could float on latest by default and allow conservative customers to pin, or we could pin and have to manually test/bump the world a couple times a year.  We might want to talk offline with Jeff Richter and Mike Kistler from Azure's Breaking Change Review Board for their advice on how dangerous it would be for customers to float on latest.

Reply:
I've decided to define these for all resources and put them in a shared file. Some things, like KeyVault, need to be used across many libraries. Thoughts on this approach?

Note I still need to ensure all resources use this pattern. Right now EventHubs doesn't.

Reply:
I prefer updating the library and letting that determine the version of the default azure resource. If we don't do that then this will be just like container image tags except it'll be managed in both the Azure.Provisioning libraries and in aspire.

Reply:
I would prefer that too, but there are problems with that approach.

1. Not all resources have a default version in the CDK right now. Especially "child" resources. For those we need to provide a version, or else we get errors that it is unset.
2. The default version for some things don't actually work. For example, the CDK doesn't emit an empty "properties" objects, but the default version for ServiceBus topics (I think it was this one) says that you must have a `properties` object. I can look up the exact error, if someone cares.
3. The concern that updating to a new version will break something. @tg-msft assures me that there are breaking changes that happen from time to time. So if we just take the version that the CDK uses, and someone updates their CDK to a new version, with new default resource versions, there is a potential for our bicep to be broken.

Reply:
1 is a bug that will be fixed by either generating the same default as parent resources or using a resolver to copy those.  There's still a bit to design around the versioning story.  I'd like to to have one list of versions per library, but there are some challenges there with ARM.

Let's follow up on 2.  I currently try to omit as many empty objects as possible (i.e., no reason to serialize the entire object graph with empty braces everywhere) but we can design a way to include specific empty objects where needed if using a `new ObjectExpression()` isn't already a sufficient workaround.